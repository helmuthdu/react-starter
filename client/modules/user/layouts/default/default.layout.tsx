import React, { Fragment } from 'react';

import './default.layout.scss';

// eslint-disable-next-line
export const DefaultLayout = ({ children }: { children: React.ReactNode }) => {
  return <Fragment>{children}</Fragment>;
};

export default DefaultLayout;
