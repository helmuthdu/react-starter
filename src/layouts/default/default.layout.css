@import 'https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro';

body,
html {
  padding: 0;
  margin: 0;
  overflow: hidden auto;
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 16px;
  background-color: #f1f1f1;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0 0 1rem;
}
