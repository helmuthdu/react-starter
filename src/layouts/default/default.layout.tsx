import './default.layout.scss';
import React, { Fragment } from 'react';
import { Outlet } from 'react-router-dom';

export const DefaultLayout = () => (
  <Fragment>
    <Outlet />
  </Fragment>
);

export default DefaultLayout;
