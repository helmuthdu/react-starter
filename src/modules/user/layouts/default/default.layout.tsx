import * as React from 'react';
import { Fragment } from 'react';

import './default.layout.scss';

// eslint-disable-next-line
export const DefaultLayout = ({ children }: { children: React.ReactNode }) => {
  return <Fragment>{children}</Fragment>;
};

export default DefaultLayout;
