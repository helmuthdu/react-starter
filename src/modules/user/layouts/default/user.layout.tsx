import { Fragment } from 'react';

export const UserLayout = ({ children }: { children: JSX.Element }) => {
  return <Fragment>{children}</Fragment>;
};

export default UserLayout;
