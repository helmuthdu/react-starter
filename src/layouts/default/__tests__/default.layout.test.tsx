import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';
import { DefaultLayout, DefaultLayoutRoute } from '../default.layout';

const Component = () => <p>lorem ipsum</p>;

describe('layouts -> DefaultLayout component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <DefaultLayout>
        <Component />
      </DefaultLayout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('layouts -> DefaultLayoutRoute component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <DefaultLayoutRoute element={Component} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
