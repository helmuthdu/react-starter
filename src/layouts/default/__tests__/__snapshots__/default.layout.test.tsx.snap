// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layouts -> DefaultLayout component > should match snapshot 1`] = `<DocumentFragment />`;

exports[`layouts -> DefaultLayout component should match snapshot 1`] = `<DocumentFragment />`;
